arches/app/models/graph.py

```diff
@@ -1409,18 +1409,21 @@ def serialize(self, fields=None, exclude=None, force_recalculation=False, use_ra
         exclude = [] if exclude is None else exclude
 
         if self.publication and not force_recalculation:
-            user_language = translation.get_language()
-            published_graph = models.PublishedGraph.objects.get(publication=self.publication, language=user_language)
-
-            serialized_graph = None
-            if published_graph:
+            try:
+                user_language = translation.get_language()
+                published_graph = models.PublishedGraph.objects.get(publication=self.publication, language=user_language)
                 serialized_graph = published_graph.serialized_graph
+                for key in exclude:
+                    if serialized_graph.get(key) is not None:  # explicit None comparison so falsey values will still return
+                        serialized_graph[key] = None
+
+                return serialized_graph
+            except:
+                self.refresh_from_database()
+                return self.serialize(
+                    fields=fields, exclude=exclude, force_recalculation=True, use_raw_i18n_json=use_raw_i18n_json, **kwargs
+                )
```

Comment:
If `self.refresh_from_database()` fails, this will enter an infinite loop

Reply:
Because I set `force_recalculation=True`, it should never enter the `if` part of that method and so, shouldn't ever get into an infinite loop scenario.  I'm not sure what would happen if it doesn't successfully refresh from the db.  Didn't test that.  Probably nothing good....  :(
Let me know if I'm missing something.

Reply:
Ya fair actually that makes sense 👍